File: telemetry/server/src/openmct/object_types.rs
use serde::Serialize;

#[derive(Serialize, Clone)]
pub struct OpenMctObject {
    pub id: &'static str,
    pub name: &'static str,
    pub icon: &'static str,
}

pub static OPEN_MCT_OBJECT_TYPES: &'static [OpenMctObject] = &[
    OpenMctObject {
        id: "temperature",
        name: "Temperature",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "thermistor",
        name: "Thermistor",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "acceleration",
        name: "Acceleration",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "pressure",
        name: "Pressure",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "hall_effect",
        name: "Hall Effect",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "displacement",
        name: "Displacement",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "velocity",
        name: "Velocity",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "status",
        name: "Status",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "keyence",
        name: "Keyence",
        icon: "icon-telemetry",
    },
    OpenMctObject {
        id: "brake_feedback",
        name: "Brake Feedback",
        icon: "icon-telemetry",
    },
];

File: telemetry/packages/constants/src/openmct/object-types/object-types.ts
import type { OpenMctObjectTypes } from '@hyped/telemetry-types';
import type { telemetryTypes } from '../../pods/types';

export type OpenMctObjectTypeId = (typeof telemetryTypes)[number];

type StrictOpenMctObjectTypes = OpenMctObjectTypes &
	{ id: OpenMctObjectTypeId }[];

export const openMctObjectTypes: StrictOpenMctObjectTypes = [
	{
		id: 'temperature',
		name: 'Temperature',
		icon: 'icon-telemetry',
	},
	{
		id: 'acceleration',
		name: 'Acceleration',
		icon: 'icon-telemetry',
	},
	{
		id: 'pressure',
		name: 'Pressure',
		icon: 'icon-telemetry',
	},
	{
		id: 'hall_effect',
		name: 'Hall Effect',
		icon: 'icon-telemetry',
	},
	{
		id: 'displacement',
		name: 'Displacement',
		icon: 'icon-telemetry',
	},
	{
		id: 'velocity',
		name: 'Velocity',
		icon: 'icon-telemetry',
	},
	{
		id: 'status',
		name: 'status',
		icon: 'icon-telemetry',
	},
	{
		id: 'magnetism',
		name: 'Magnetism',
		icon: 'icon-telemetry',
	},
	{
		id: 'keyence',
		name: 'Keyence',
		icon: 'icon-telemetry',
	},
	{
		id: 'resistance',
		name: 'Resistance',
		icon: 'icon-telemetry',
	},
	{
		id: 'levitation',
		name: 'Levitation',
		icon: 'icon-telemetry',
	},
	{
		id: 'binary-status',
		name: 'Binary Status',
		icon: 'icon-telemetry',
	},
];
